### PR TITLE
Distortion/LensFlarePlugin: set compositor names

### LINKED ISSUE #3005: Allow specifying lens flare and camera distortion texture format
Two possible solutions:
* Change the hard-coded values from `PF_A8R8G8B8` to `PF_FLOAT32_RGBA`. This will could impact performance for all users but is the easiest change.
* Add an API for specifying the name of the compositor file in the lens flare and distortion classes. Then users can add their own compositor and change the pixel format as needed.

### PR DESCRIPTION
**Pull request by audrow**
Fixes #3005 by taking the second approach suggested.

### CODE REVIEW

**Comment by scpeters** [comment]
the homebrew build is failing due to https://github.com/osrf/homebrew-simulation/issues/1477 / https://github.com/osrf/gazebo/issues/2867

File: test/integration/camera_sensor.cc [hunk 1]
```diff
@@ -1929,3 +1929,112 @@ TEST_F(CameraSensor, Light)
...
+TEST_F(CameraSensor, SetCompositorNames)
+{
+#if not defined(__APPLE__)
+  Load("worlds/test/issue_3005_set_compositor_names.world");
+
+  // Make sure the render engine is available.
+  if (rendering::RenderEngine::Instance()->GetRenderPathType() ==
+      rendering::RenderEngine::NONE)
+  {
+    gzerr << "No rendering engine, unable to run wide angle camera test\n";
```
Comment (scpeters) [inline test/integration/camera_sensor.cc]: nit: change "wide angle camera test" to  something more suitable

File: test/integration/camera_sensor.cc [hunk 2]
```diff
@@ -1929,3 +1929,112 @@ TEST_F(CameraSensor, Light)
...
+    return;
+  }
+
+  const unsigned int width  = 12;
+  const unsigned int height = 12;
+
+  for (auto camera_name : {"camera_distortion_default", "camera_lens_flare_default"})
```
Comment (scpeters) [inline test/integration/camera_sensor.cc]: nit: I think this line is longer than 80 characters

File: test/integration/camera_sensor.cc [hunk 3]
```diff
@@ -1929,3 +1929,112 @@ TEST_F(CameraSensor, Light)
...
+  {  // check camera with default texture format
```
Comment (scpeters) [inline test/integration/camera_sensor.cc]: nit: move comment to separate line from `{`

File: test/integration/camera_sensor.cc [hunk 4]
```diff
@@ -1929,3 +1929,112 @@ TEST_F(CameraSensor, Light)
...
+    EXPECT_NE(gSum, bSum);
+
+    EXPECT_GT(rSum - bSum, 25000.0);
+    EXPECT_GT(rSum - gSum, 20000.0);
+
+    delete [] img;
+  }
+
+  for (auto camera_name : {"camera_distortion_test", "camera_lens_flare_test"})
+  {  // check camera with grayscale texture format
```
Comment (scpeters) [inline test/integration/camera_sensor.cc]: nit: move comment to separate line from `{`

**Review by scpeters — COMMENTED** [review]
this looks great overall; I just added a few minor comments related to code style. Some of them may have been caught by `sh tools/code_check.sh` (run from the root of the repository), but there are so many existing failures, that it can be hard to identify new errors, so I wouldn't require you to do that

**Comment by scpeters** [comment]
I expect the homebrew build will fail again until https://github.com/ignition-tooling/release-tools/pull/465 is merged

File: test/integration/camera_sensor.cc (hunk 1 above)
Comment (audrow) [reply test/integration/camera_sensor.cc]: Good catch!

**Review by scpeters — APPROVED** [review]

**Comment by scpeters** [comment]
I think the homebrew CI will work now, so I pushed a small fix to a test that has been failing for a long time to restart CI.

**Review by WilliamLewww — APPROVED** [review]

**Pull request merged**